File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
  def jump(self, nums: List[int]) -> int:
    ans = l = r = ub = 0
    while r < len(nums)-1:
      for i in range(l, r+1):
        ub = max(ub, i+nums[i])
      l = r+1
      r = ub
      ans += 1
    return ans

File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
  public int jump(int[] nums) {
    int ans = 0;
    int l = 0;
    int r = 0;
    int ub = 0;
    while (r < nums.length - 1) {
      for (int i = l; i <= r; i++) {
        ub = Math.max(ub, i + nums[i]);
      }
      l = r + 1;
      r = ub;
      ans++;
    }
    return ans;
  }
}